src/Tooltip/Tooltip.tsx

```diff
@@ -7,84 +7,124 @@ import React, {
   useState,
 } from 'react'
 import styled, { css } from 'styled-components'
+import { useEventListener } from '../hooks'
 import { Text } from '../Text'
 import { theme } from '../theme'
 
 type Position = 'top' | 'bottom' | 'left' | 'right'
-type Size = 'small' | 'large'
-type ArrowPosition = 'left' | 'right' | 'center' | 'top' | 'bottom'
+type ArrowPosition = Position | 'center'
 
 export interface TooltipProps {
   children: ReactNode
   title?: string
   content: string | ReactNode
   position: Position
   underline?: boolean
-  size?: Size
-  arrowPosition?: ArrowPosition
+  defaultArrowPosition?: ArrowPosition
   shadow?: boolean
 }
 
 export const Tooltip: FC<TooltipProps> = ({
   children,
   title,
   content,
-  position = 'top',
   underline = false,
-  size = 'small',
-  arrowPosition = 'center',
+  defaultArrowPosition = 'center',
   shadow = false,
 }) => {
   const tipContainer = useRef<HTMLDivElement>(null)
-  const [tooltipPosition, setTooltipPosition] = useState<Position>(position)
+  const documentRef = useRef<Document>(document)
+  const [tooltipPosition, setTooltipPosition] = useState<Position>('top')
+  const [arrowPosition, setArrowPosition] =
+    useState<ArrowPosition>(defaultArrowPosition)
 
-  const checkInbounds = (element: DOMRect) =>
+  const checkInbounds = (element: DOMRect): boolean =>
     element.top >= 0 &&
     element.left >= 0 &&
     element.bottom <= window.innerHeight &&
     element.right <= window.innerWidth
 
   const handleTipViewport = useCallback(() => {
+    const shouldChangeXAxis: boolean =
+      tooltipPosition === 'left' || tooltipPosition === 'right'
+    const shouldChangeYAxis: boolean =
+      tooltipPosition === 'top' || tooltipPosition === 'bottom'
```

Comment:
```suggestion
    const shouldChangeXAxis =
      tooltipPosition === 'left' || tooltipPosition === 'right'
    const shouldChangeYAxis =
      tooltipPosition === 'top' || tooltipPosition === 'bottom'
```

the type should be inferred, right?

Reply:
Have updated now 😄